File: vol1/ch8/LeetCode108.py
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
         return self.preorder(nums,0,len(nums)-1)

    def preorder(self,nums,low,high):
        if low>high: return None
        mid=(low+high)//2
        root=TreeNode(nums[mid])
        root.left=self.preorder(nums,low,mid-1)
        root.right=self.preorder(nums,mid+1,high)
        return root

File: vol1/ch8/LeetCode108.cpp
class Solution {
public:
    TreeNode* sortedArrayToBST(vector<int>& nums) {
        return preorder(nums,0,nums.size()-1);
    }

    TreeNode* preorder(vector<int>&nums,int low,int high) {
        if(low>high) return NULL;
        int mid=(low+high)/2;
        TreeNode*root=new TreeNode(nums[mid]);
        root->left=preorder(nums,low,mid-1);
        root->right=preorder(nums,mid+1,high);
        return root;
    }
};

